tools/nodejs_api/src_cpp/node_connection.cpp

```diff
@@ -41,6 +41,8 @@ Napi::Value NodeConnection::InitAsync(const Napi::CallbackInfo& info) {
 
 void NodeConnection::InitCppConnection() {
     this->connection = std::make_shared<Connection>(database.get());
+    connection->getClientContext()->getProgressBar()->setDisplay(
+        std::make_shared<NodeProgressBarDisplay>());
```

Comment:
not related to this PR. But I think you might be able to use unique_ptr for display in ProgressBar class.